### PR TITLE
Refactor EnumerateChildren to avoid need for bestEffort paramater.

### PR DESCRIPTION
**Pull request by kevina**
License: MIT
Signed-off-by: Kevin Atkinson <k@kevina.org>

### CODE REVIEW

**Comment by kevina** [comment]
WIP.  A test is failing locally for me and I want to see if it okay on the build servers.

**Comment by kevina** [comment]
@whyrusleeping okay the tests are passing.  Let me know if you are like this approach.  If you do I will refactor the Async version in a similar fashion.

**Comment by kevina** [comment]
> Lets make sure not to change the function parameter ordering. the new function parameter should take the place of the linkservice

I only did that because I thought the new order made more sense.  Is there a reason you don't like it?  Just curious.  I have no problem changing them back.

**Comment by whyrusleeping** [comment]
@kevina Eh, I think it makes more sense to have the fetcher first, as its almost as if the cid could be an argument to a method on the 'fetcher function'.

**Comment by kevina** [comment]
> @kevina Eh, I think it makes more sense to have the fetcher first, as its almost as if the cid could be an argument to a method on the 'fetcher function'.

Okay, will keep the order the same.  Will also see what I can do to refactor the Async version, although that will be a little more involved.

File: merkledag/merkledag.go [hunk 1]
```diff
@@ -36,8 +36,10 @@ type DAGService interface {
 }
 
 type LinkService interface {
-	// Return all links for a node, may be more effect than
-	// calling Get in DAGService
+	// Return all links for a node.  The complete node does not
+	// necessarily have to exist locally, or at all.  For example, raw
+	// leaves can not possible have links so there is no need to look
```
Comment (whyrusleeping) [inline merkledag/merkledag.go]: typo: cannot possibly

File: merkledag/merkledag.go [hunk 2]
```diff
@@ -114,6 +116,8 @@ func decodeBlock(b blocks.Block) (node.Node, error) {
 	}
 }
 
+// GetLinks return the links for the node, the node doesn't necessary
```
Comment (whyrusleeping) [inline merkledag/merkledag.go]: typo: necessarily

File: merkledag/merkledag.go [hunk 3]
```diff
@@ -138,9 +142,22 @@ func (n *dagService) Remove(nd node.Node) error {
 	return n.Blocks.DeleteBlock(nd)
 }
 
+// Create a functtion to get the links for a node, from the node,
```
Comment (whyrusleeping) [inline merkledag/merkledag.go]: typo: function, guaranteed

Comment (Kubuxu) [inline merkledag/merkledag.go]: Also go style of docs says that you should start doc comment with the name of the function.

**Comment by kevina** [comment]
@Kubuxu I fixed my comments to include the start with the name of the function

@whyrusleeping I corrected the typos you pointed out.

**Comment by whyrusleeping** [comment]
Alright cool, this is ready to go in 0.4.7

**Comment by whyrusleeping** [comment]
@kevina got a merge conflict here. Mind rebasing?

**Comment by kevina** [comment]
@whyrusleeping sorry, missed this.  Just rebased and things should be good to go.

**Comment by whyrusleeping** [comment]
Thanks @kevina !

**Pull request merged**